src/lib/node.ts

```diff
@@ -209,3 +209,75 @@ export function pipeToNodeStream<T>(
 		handleTsStreamError // abort handler
 	);
 }
+
+/**
+ * Create a ts-stream ReadableStream from a Node.JS ReadableStream.
+ *
+ * Reads all values from `nodeReadable` and writes them to the returned stream.
+ * When the `nodeReadable` ends, the returned stream is also ended.
+ * Should the `nodeReadable` emit a `close` event, the returned stream is ended without error.
+ * If the `nodeReadable` emits a `error` event, the returned stream is ended with the given error.
+ *
+ * Usage example:
+ * ```ts
+ * const source = fromNodeReadable(fs.createReadStream('source.txt'));
+ * source
+ *  .map((chunk) => String(chunk))
+ *  .map((letters) => letters.toUpperCase())
+ *  .forEach(console.log);
+ * ```
+ *
+ * @param nodeReadable Soource NodeJS stream
+ * @returns Readable stream of type `string | Buffer`
+ */
+export function fromNodeReadable(
+	nodeReadable: NodeJS.ReadableStream
+): ReadableStream<string | Buffer> {
+	const stream = new Stream<string | Buffer>();
+
+	const endStream = async (err?: Error) => {
+		if (!stream.isEndingOrEnded()) return stream.end(err);
+	};
+
+	stream.aborted().catch(endStream);
```

Comment:
Does it make sense to also destroy the readable with the abort error?

Reply:
Just to clarify, are you asking if we should emit an `error` event on behalf of the `nodeReadable` stream?

Reply:
Almost, I meant to call its `destroy()`.